### ISSUE TITLE
QST:  how to use groupby.rolling correctly? Please help. Not sure what I am doing wrong

### ISSUE THREAD

**Issue by terrygryffindor** [pandas-dev/pandas#41078]
- [x] I have searched the [[pandas] tag](https://stackoverflow.com/questions/tagged/pandas) on StackOverflow for similar questions.

- [x] I have asked my usage related question on [StackOverflow](https://stackoverflow.com).

---

#### Question about pandas

Hi there, my understanding of the df.groupby.rolling for a multi index dataframe should preserve the level-1 index, but I can't seem to get that. I am running Pandas 1.2.4. Hope that somebody can provide some insight of what is happening

```python
x = range(0, 6)
id = ['a', 'a', 'a', 'b', 'b', 'b']
temp = pd.DataFrame(zip(id, x), columns = ['id', 'x'])
temp['sequence']=[0,1,2,0,1,2]
temp=temp.set_index(['id','sequence'])
print(temp)

             x
id sequence   
a  0         0
   1         1
   2         2
b  0         3
   1         4
   2         5

temp.groupby(level=0,group_keys=True).rolling(2, min_periods=1).min()

      x
id     
a   0.0
a   0.0
a   1.0
b   3.0
b   3.0
b   4.0



```
base don the online guides, I would expect the level 1 index to be preserved...but I can't understand what is currently happening at this case? or what is the correct way of doing it.

Hope that somebody can provide some insights with regards to this problem

EDIT: Edited as text for readability, apologies for posting images earlier, forgot about it while posting it just now

**Comment by phofl** [comment]
Could you provide your example as "text" not as a screenshot?

@mroeschke Is this documented somewhere? Iirc this changed recently/is a current discussion topic how this should behave

**Comment by mroeschke** [comment]
Yes this was fixed recently by https://github.com/pandas-dev/pandas/pull/40701 and will be fixed in the 1.3 release. Unfortunately this has been broken since the 1.1.3 release.

```
In [1]: x = range(0, 6)
   ...: id = ['a', 'a', 'a', 'b', 'b', 'b']
   ...: temp = pd.DataFrame(zip(id, x), columns = ['id', 'x'])
   ...: temp['sequence']=[0,1,2,0,1,2]
   ...: temp=temp.set_index(['id','sequence'])
   ...: print(temp)
             x
id sequence
a  0         0
   1         1
   2         2
b  0         3
   1         4
   2         5

In [2]: temp.groupby(level=0,group_keys=True).rolling(2, min_periods=1).min()
Out[2]:
                  x
id id sequence
a  a  0         0.0
      1         0.0
      2         1.0
b  b  0         3.0
      1         3.0
      2         4.0

In [3]: pd.__version__
Out[3]: '1.3.0.dev0+1392.g11545d5023'
```

Thanks for the report. Going to close as fixed in a future release.
